src/main/java/hudson/plugins/s3/utils/S3Utils.java

```diff
@@ -0,0 +1,64 @@
+package hudson.plugins.s3.utils;
+
+import hudson.ProxyConfiguration;
+import hudson.util.Secret;
+
+import java.util.regex.Pattern;
+
+import jenkins.model.Jenkins;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.internal.StaticCredentialsProvider;
+import com.amazonaws.services.s3.AmazonS3Client;
+
+public class S3Utils {
+
+    public static AmazonS3Client createClient(String accessKey, Secret secretKey, boolean useRole, String stsRoleArn) {
+        AWSCredentialsProvider credentialsProvider;
+        if (useRole) {
+            credentialsProvider = new DefaultAWSCredentialsProviderChain();
```

Comment:
Ok. We could just copy the private class `S3CredentialsProviderChain` and use it the same way. A bit ugly, but it should work.

Unfortunately, I don't have an AWS account anymore, so I can't test this.